install/kubernetes/cilium/templates/cilium-nodeinit-daemonset.yaml

```diff
@@ -242,8 +242,25 @@ spec:
                   docker kill "${container_id}" || true
                 done
               else
-                # COS-beta (with containerd). Some versions of COS have crictl in /home/kubernetes/bin.
-                for f in `find /var/lib/cni/networks/ -type f ! -name lock ! -name last_reserved_ip.0`; do PATH="${PATH}:/home/kubernetes/bin" crictl stopp "$(sed 's/\r//;1q' $f)" || true; done
+                if [ -d /var/lib/cni/networks/ ]; then
+                    echo "Sleeping for 60 seconds..."
+                    sleep 60
+                    echo "/var/lib/cni/networks/ found. Checking files for container-ids which are not managed by cilium ..."
+                    # COS-beta (with containerd). Some versions of COS have crictl in /home/kubernetes/bin. 
+                    for f in `find /var/lib/cni/networks/ -type f ! -name lock ! -name last_reserved_ip.0`; do PATH="${PATH}:/home/kubernetes/bin" crictl stopp "$(sed 's/\r//;1q' $f)" || true; done
+                elif [ -d /var/lib/cni/cache/results/ ]; then
+                    # AKS 1.18.X uses /var/lib/cni/cache/results and docker
+                    echo "Sleeping for 60 seconds..."
+                    sleep 60
+                    echo "/var/lib/cni/cache/results/ found. Checking files for container-ids which are not managed by cilium ..."
+                    for f in `find /var/lib/cni/cache/results/ -type f ! -name "cilium-*" ! -name "cni-loopback-*" -printf "%f\n"`; do IFS="-" read -a id <<<"$f" ; docker kill ${id[1]} || true; done
```

Comment:
What do the contents of the files in `/var/lib/cni/cache/results/` look like? That would be nice to include as comments here, since getting access to them is not so easy. Which transformation are you performing using the `read`?

Reply:
I will write a summary for all AKS-Versions (1.18.X / 1.19.X / 1.20.X) with my next push.